### PR TITLE
[FEATURE] Make Screenshots of the EXT:examples frontend

### PR DESCRIPTION
**Pull request by linawolf**
Solves https://github.com/TYPO3-Documentation/t3docs-screenshots/issues/164

### CODE REVIEW

File: .ddev/config.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-name: t3docs-screenshots
+name: sig-screen-examples
```
Comment (alexander-nitsche) [inline .ddev/config.yaml:1]: Regression ..

File: .ddev/config.yaml [hunk 2]
```diff
@@ -64,7 +64,7 @@ composer_version: "2"
 # see https://en.wikipedia.org/wiki/List_of_tz_database_time_zones
 # For example Europe/Dublin or MST7MDT
 
-# composer_version: "2"
+# composer_version: ""
```
Comment (alexander-nitsche) [inline .ddev/config.yaml:67]: Does it use composer v2 by default now?

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 3]
```diff
@@ -221,7 +221,34 @@ public function createBasicConfig(): void
...
-                        [
+                        "_default"=> [
+                            ["action"=> "resizeWindow", "width"=> 640,"height"=> 640],
+                            ["action"=> "setScreenshotsDocumentationPath", "path"=> "Documentation"],
+                            ["action"=> "setScreenshotsImagePath","path"=> "Images/AutomaticScreenshots"],
+                            ["action"=> "setScreenshotsRstPath","path"=> "Images/AutomaticScreenshots"],
+                            ["action"=> "setCodeSnippetsSourcePath","path"=> "typo3conf/ext/examples"],
+                            ["action"=> "setCodeSnippetsTargetPath", "path"=> "CodeSnippets"]
+                        ],
+                        "frontend" => [
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:232]: Keep prefix "actionsIdentifier.." for all identifiers in the MyManual. See other identifiers.

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 4]
```diff
@@ -221,7 +221,34 @@ public function createBasicConfig(): void
...
+                            ["include" => "_default"],
+                            [
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:234]: Keep one action per one line.

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 5]
```diff
@@ -221,7 +221,34 @@ public function createBasicConfig(): void
                 ],
                 'Examples' => [
                     'screenshots' => [
-                        [
+                        "_default"=> [
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:224]: Remove that _default section as it does not provide additional insights. This is copied from a production screenshots.json but has no additional benefit here.

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 6]
```diff
@@ -221,7 +221,34 @@ public function createBasicConfig(): void
...
+                            [
+                                "action" => "resizeWindow",
+                                "width" => 480,
+                                "height" => 640
+                            ],
+                            [
+                                "action" => "amOnPage",
+                                "page" => "/?id=1"
+                            ],
+                            ["action" => "wait", "timeout" => 1],
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:243]: Prefer `waitForText` or `waitForElement` over `wait` ..

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 7]
```diff
@@ -221,7 +221,34 @@ public function createBasicConfig(): void
...
+                            [
+                                "action" => "makeScreenshotOfElement",
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:245]: Prefer `makeScreenshotOfWindow` over `makeScreenshotOfElement` if no selector is specified ..

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 8]
```diff
@@ -270,6 +297,64 @@ public function createBasicConfig(): void
                             ['action' => 'click', 'link' => "List"],
                             ['action' => 'makeScreenshotOfFullPage', 'fileName' => "IntroductionDashboardFullPage"],
                         ],
+                        "startpage" => [
+                            [
+                                "action" => "setScreenshotsImagePath",
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:302]: Superfluous ..

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 9]
```diff
@@ -270,6 +297,64 @@ public function createBasicConfig(): void
...
+                                "path" => "Images/AutomaticScreenshots/IntroductionPackage"
+                            ],
+                            [
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:305]: Superfluous ..

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 10]
```diff
@@ -270,6 +297,64 @@ public function createBasicConfig(): void
...
+                                "action" => "setScreenshotsRstPath",
+                                "path" => "Images/AutomaticScreenshots/IntroductionPackage"
+                            ],
+                            [
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:309]: Superfluous ..

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 11]
```diff
@@ -221,7 +221,34 @@ public function createBasicConfig(): void
...
+                            ["action"=> "setScreenshotsDocumentationPath", "path"=> "Documentation"],
+                            ["action"=> "setScreenshotsImagePath","path"=> "Images/AutomaticScreenshots"],
+                            ["action"=> "setScreenshotsRstPath","path"=> "Images/AutomaticScreenshots"],
+                            ["action"=> "setCodeSnippetsSourcePath","path"=> "typo3conf/ext/examples"],
+                            ["action"=> "setCodeSnippetsTargetPath", "path"=> "CodeSnippets"]
+                        ],
+                        "frontend" => [
+                            ["include" => "_default"],
+                            [
+                                "action" => "resizeWindow",
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:235]: Superfluous ..

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 12]
```diff
@@ -270,6 +297,64 @@ public function createBasicConfig(): void
                             ['action' => 'click', 'link' => "List"],
                             ['action' => 'makeScreenshotOfFullPage', 'fileName' => "IntroductionDashboardFullPage"],
                         ],
+                        "startpage" => [
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:300]: Remove the former EXT:introduction backend screenshot section as the new backend section does the same. Split EXT:introduction backend and frontend screenshot into two sections with own identifier, as done for EXT:examples.

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 13]
```diff
@@ -270,6 +297,64 @@ public function createBasicConfig(): void
...
+                                "path" => ["Congratulations"]
+                            ],
+                            ["action" => "switchToContentFrame"],
+                            [
+                                "action" => "waitForText",
+                                "text" => "Congratulations",
+                                "timeout" => 5
+                            ],
+                            [
+                                "action" => "makeScreenshotOfElement",
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:333]: Prefer `makeScreenshotOfWindow` over `makeScreenshotOfElement` if no selector is specified ..

File: packages/screenshots/Classes/Configuration/Configuration.php [hunk 14]
```diff
@@ -270,6 +297,64 @@ public function createBasicConfig(): void
...
+                                "action" => "resizeWindow",
+                                "width" => 1024,
+                                "height" => 2048
+                            ],
+                            [
+                                "action" => "amOnPage",
+                                "page" => "/?id=1"
+                            ],
+                            [
+                                "action" => "wait",
```
Comment (alexander-nitsche) [inline packages/screenshots/Classes/Configuration/Configuration.php:347]: Prefer `waitForText` or `waitForElement` over `wait` ..

**Review by alexander-nitsche — CHANGES_REQUESTED** [review]

File: packages/screenshots/Classes/Configuration/Configuration.php (hunk 3 above)
Comment (linawolf) [reply packages/screenshots/Classes/Configuration/Configuration.php:232]: Whats the point of using such long and complicated names?

File: packages/screenshots/Classes/Configuration/Configuration.php (hunk 4 above)
Comment (linawolf) [reply packages/screenshots/Classes/Configuration/Configuration.php:234]: actually if you would use auto formating of the php file with our current .editorconfig all lines would look like mine not like yours.... I had my lines automatically formated

File: packages/screenshots/Classes/Configuration/Configuration.php (hunk 5 above)
Comment (linawolf) [reply packages/screenshots/Classes/Configuration/Configuration.php:224]: In my eyes it has the same benefit like in the introduction section. This should be the standard way of doing these kind of settings.

File: packages/screenshots/Classes/Configuration/Configuration.php (hunk 14 above)
Comment (linawolf) [reply packages/screenshots/Classes/Configuration/Configuration.php:347]: This is  a demontstration of how to take a screenshot of the frontend. It shouldnt depend on text to be found.

File: packages/screenshots/Classes/Configuration/Configuration.php (hunk 13 above)
Comment (linawolf) [reply packages/screenshots/Classes/Configuration/Configuration.php:333]: Why? Make screenshotOfWindow only seems to make sense when you want to catchOverlays or the such

File: .ddev/config.yaml (hunk 2 above)
Comment (linawolf) [reply .ddev/config.yaml:67]: It depends on your setup. But as our packages require bootstrap v2 it is better to require it explicitely

**Review by alexander-nitsche — APPROVED** [review]
Changes applied. My local My-Manual run failed due to failing distribution imports. Test on your side a last time to ensure it works fine.

**Comment by linawolf** [comment]
Works fine for me, thank you!

**Pull request merged**